gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/jdbc/EmbedStatement.java

```diff
@@ -1667,13 +1691,16 @@ public final void clearBatch() throws SQLException {
 			Vector stmts = batchStatements;
 			*/
 // GemStone changes END
-			batchStatements = null;
+			if (stmts != null && (stmts.size() > 0) && stmts.get(0) instanceof String){
+				batchStatements = null;
+			}
+
 			int size;
 			if (stmts == null)
 				size = 0;
 			else
-				size = stmts.size();
-
+				size = batchStatementCurrentIndex;//stmts.size();
```

Comment:
This looks problematic because batchStatementCurrentIndex is not being updated for the case "executeBatchInProgress == 0" in EmbedPreparedStatement. I think you will need to update batchStatementCurrentIndex in the getClone() case too.